src/dmd/root/speller.d

```diff
@@ -239,25 +239,26 @@ unittest
     ];
     //printf("unittest_speller()\n");
 
-    void* dgarg;
+    string dgarg;
 
-    void* speller_test(const(char)* s, ref int cost)
+    void* speller_test(const(char)[] s, ref int cost)
     {
+        assert(s[$-1] != '\0');
         //printf("speller_test(%s, %s)\n", dgarg, s);
         cost = 0;
-        if (strcmp(cast(char*)dgarg, s) == 0)
-            return dgarg;
+        if (dgarg == s)
+            return cast(void*)dgarg;
```

Comment:
This is casting away immutable... That should never happen.
From a quick look, couldn't `speller_test` return `const(void)*` ?

Reply:
You're right, but other users of speller expect the delegate to return non-const. I figured it was better to let one unittest cast away immutable than let the actual program cast away const. I think a solution to this problem would be to make the delegate a template parameter so we don't need the cast at all.